### DISCUSSION TITLE
Ability to clear app cache

### DISCUSSION THREAD

**Discussion by SoCuul** [arichornlover/uYouEnhanced#320] [Ideas]
The YouTube app tends to cache a fair amount of data, to the point where it becomes a pain to use on devices with lower amounts of storage.

If we could have an option in settings to clear the cache folders without reinstalling the app, that would be amazing!

**Comment by ego-lay-atman-bay** [comment]
Does your OEM not allow you to delete cache?
![Screenshot_20240130-060235](https://github.com/arichornlover/uYouEnhanced/assets/70973578/92780865-839e-446a-84a0-7533ef271381)

**Reply by yodaluca23** [reply]
> Does your OEM not allow you to delete cache?
> 
> ![Screenshot_20240130-060235](https://github.com/arichornlover/uYouEnhanced/assets/70973578/92780865-839e-446a-84a0-7533ef271381)

No, that not a thing in IOS, and this is an IOS app, so I don't see how bringing up an Android OS feature while your using a completely different app, is relevant...

**Reply by ego-lay-atman-bay** [reply]
Oh yeah, oops, for some reason I thought this was the revanced repo.

**Reply by SoCuul** [reply]
To be fair Apple should allow this in settings. Would make everything so much easier.

**Comment by arichornlover** [comment]
It’s possible to clear the app cache on iOS. The feature exists in both YTMusicUltimate & YouTubeRebornPlus. But however, the clear cache button depends on using UITableView. How can I port it into YouTube Settings?

**Reply by SoCuul** [reply]
Can't you just implement it like you've implemented everything else? The logic isn't what's based on UITableView.

**Reply by arichornlover** [reply]
https://github.com/arichornlover/uYouEnhanced/commit/c2510652f0e896f087c08564187bc8d8bfdafd5e
I did try to port it. Do you think this would work?

**Comment by SoCuul** [comment]
> [c251065](https://github.com/arichornlover/uYouEnhanced/commit/c2510652f0e896f087c08564187bc8d8bfdafd5e) I did try to port it. Do you think this would work?

I'm unsure personally, as I am nowhere proficient in Objective-C. Maybe try it on your device and see if it works!
